kubernetes/resource_kubernetes_persistent_volume_v1_test.go

```diff
@@ -0,0 +1,2000 @@
+    persistent_volume_source {
+      azure_disk {
+        caching_mode  = "None"
+        data_disk_uri = %[2]q
+        disk_name     = %[1]q
+      }
+    }
+  }
+}`, name, dataDiskURI)
+}
+
+func testAccKubernetesPersistentVolumeV1Config_azure_PersistentVolumeKind(name, dataDiskURI, kind string) string {
+	return fmt.Sprintf(`resource "kubernetes_persistent_volume_v1" "test" {
+  metadata {
+    name = %[1]q
+  }
+  spec {
+    capacity = {
+      storage = "1Gi"
+    }
+    access_modes = ["ReadWriteOnce"]
+    persistent_volume_source {
+      azure_disk {
+        caching_mode  = "None"
+        data_disk_uri = %[2]q
+        disk_name     = %[1]q
+        kind          = %[3]q
+      }
+    }
+  }
+}`, name, dataDiskURI, kind)
+}
+
+func testAccKubernetesPersistentVolumeV1Config_azure_blobStorage(name, location string) string {
+	return fmt.Sprintf(`provider "azurerm" {
+  features {}
+}
+resource "azurerm_resource_group" "test" {
+  name     = "%s"
+  location = "%s"
+  tags = {
+    environment = "terraform-provider-kubernetes-test"
+  }
+}
+resource "azurerm_storage_account" "test" {
+  name                     = %[1]q
+  resource_group_name      = azurerm_resource_group.test.name
+  location                 = azurerm_resource_group.test.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+  account_kind             = "BlobStorage"
+  tags = {
+    environment = "terraform-provider-kubernetes-test"
+  }
+}
+resource "azurerm_storage_container" "test" {
+  name                  = %[1]q
+  storage_account_name  = azurerm_storage_account.test.name
+  container_access_type = "private"
+}
+`, name, location)
+}
+
+func testAccKubernetesPersistentVolumeV1Config_azure_PersistentVolumeAzureFile(name, secretName string) string {
+	return fmt.Sprintf(`resource "kubernetes_persistent_volume" "test" {
+  metadata {
+    name = %[1]q
+  }
+  spec {
+    capacity = {
+      storage = "1Gi"
+    }
+    access_modes = ["ReadWriteOnce"]
+    persistent_volume_source {
+      azure_file {
+        secret_name = %[2]q
+        share_name  = %[1]q
+        read_only   = false
+      }
+    }
+  }
+}`, name, secretName)
+}
+
+func testAccKubernetesPersistentVolumeV1Config_azure_PersistentVolumeAzureFileNamespace(name, namespace, secretName string) string {
+	return fmt.Sprintf(`resource "kubernetes_namespace" "test" {
+  metadata {
+    name = %[2]q
+  }
+}
+
+resource "kubernetes_secret" "test" {
+  metadata {
+    name      = %[3]q
+    namespace = %[2]q
+  }
+
+  data = {
+    azurestorageaccountname = azurerm_storage_account.test.name
+    azurestorageaccountkey  = azurerm_storage_account.test.primary_access_key
+  }
+}
+
+resource "kubernetes_persistent_volume_v1" "test" {
+  metadata {
+    name = %[1]q
+  }
+  spec {
+    capacity = {
+      storage = "1Gi"
+    }
+    access_modes = ["ReadWriteOnce"]
+    persistent_volume_source {
+      azure_file {
+        secret_name      = %[3]q
```

Comment:
Could you please add here a TODO comment? We will fix this later when work on Azure-specific tests.

```
// TODO
// Use resource reference here to build dependency.
```